firebase-firestore/src/androidTest/java/com/google/firebase/firestore/TransactionTest.java

```diff
@@ -480,18 +473,50 @@ public void testCannotReadAfterWriting() {
     assertNotNull(e);
   }
 
+  @Test
+  public void testReadandUpdateNonExistentDocumentWithExternalWrite() {
+    final FirebaseFirestore firestore = testFirestore();
+    AtomicInteger counter = new AtomicInteger(0);
+
+    // Make a transaction that will fail
+    Task<Void> transactionTask =
+        firestore.runTransaction(
+            transaction -> {
+              int count = counter.incrementAndGet();
+              // Get and update a document that doesn't exist so that the transaction fails.
+              DocumentReference doc = firestore.collection("nonexistent" + count).document();
+              transaction.get(doc);
+              // Do a write outside of the transaction.
+              doc.set(map("count", count));
+              // Now try to update the other doc from within the transaction.
+              // This should fail, because the document didn't exist at the
+              // start of the transaction.
+              transaction.update(doc, "count", 1234);
+              return null;
+            });
+
+    waitForException(transactionTask);
+    assertFalse(transactionTask.isSuccessful());
+    Exception e = transactionTask.getException();
+    assertTrue(e instanceof FirebaseFirestoreException);
```

Comment:
You can drop this assertion because:

  * You're already effectively forcing this test in the line below (by casting)
  * If this were to fail this error message would be "`false` is not `true`", which is actually less informative than the `ClassCastException` you'll get on the line below.

More generally, it's important to understand how a test is failing from its error message. If you really did need to assert that a value is an instanceof some class, use a construct that more directly expresses that:

Truth style:
```java
assertThat(e).isInstanceOf(FirebaseFirestoreException.class);
```

If the file you're making the assertion in is still using JUnit/Hamcrest style, it would look like this (after static imports):
```java
assertThat(e, instanceOf(FirebaseFirestoreException.class));
```